registry/config.go

```diff
@@ -44,6 +44,10 @@ var (
 	ErrInvalidRepositoryName = errors.New("Invalid repository name (ex: \"registry.domain.tld/myrepos\")")
 
 	emptyServiceConfig = NewServiceConfig(nil)
+
+	// V2Only controls access to legacy registries.  If it is set to true via the
+	// command line flag the daemon will not attempt to contact legacy registries
+	V2Only = false
```

Comment:
how about we make this `runtime.GOOS != "linux"` that way new OSes automatically won't inherit v1.